webpack/components/extensions/HostDetails/Tabs/ModuleStreamsTab/ModuleStreamsTab.js

```diff
@@ -0,0 +1,231 @@
+  const { id: hostId, name: hostName } = useSelector(selectHostDetails);
+
+  const emptyContentTitle = __('This host does not have any Module streams.');
+  const emptyContentBody = __('Module streams will appear here when available.');
+  const emptySearchTitle = __('Your search returned no matching Module streams.');
+  const emptySearchBody = __('Try changing your search criteria.');
+  const { searchParam } = useUrlParams();
+  const columnHeaders = [
+    __('Name'),
+    __('State'),
+    __('Stream'),
+    __('Installation status'),
+    __('Installed profile'),
+  ];
+
+  const fetchItems = useCallback(
+    (params) => {
+      if (!hostId) return hostIdNotReady;
+      return getHostModuleStreams(
+        hostId,
+        params,
+      );
+    },
+    [hostId],
+  );
+
+  const response = useSelector(selectModuleStream);
+  const { results, ...metadata } = response;
+  const status = useSelector(state => selectModuleStreamStatus(state));
+  /* eslint-disable no-unused-vars */
+  const {
+    selectOne, isSelected, searchQuery, selectedCount, isSelectable,
+    updateSearchQuery, selectNone, fetchBulkParams, ...selectAll
+  } = useBulkSelect({
+    results,
+    metadata,
+    isSelectable: _result => false,
+    initialSearchQuery: searchParam || '',
+  });
+  /* eslint-enable no-unused-vars */
+
+  if (!hostId) return <Skeleton />;
+
+  const rowActions = [
+    {
+      title: __('Install'), disabled: true,
+    },
+    {
+      title: __('Update'), disabled: true,
+    },
+    {
+      title: __('Enable'), disabled: true,
+    },
+    {
+      title: __('Disable'), disabled: true,
+    },
+    {
+      title: __('Reset'), disabled: true,
+    },
+    {
+      title: __('Remove'), disabled: true,
+    },
+  ];
+
+  const EnabledIcon = ({ streamText, streamInstallStatus, upgradable }) => {
+    const INSTALLED_STATE = {
+      INSTALLED: __('Installed'),
+      UPGRADEABLE: __('Upgradable'),
+      UPTODATE: __('Up-to-date'),
+      NOTINSTALLED: __('Not installed'),
+    };
+
+    switch (true) {
+      case (streamInstallStatus?.length > 0 && streamText === 'disabled'):
+        return <TableText wrapModifier="nowrap">{INSTALLED_STATE.INSTALLED}</TableText>;
+      case (streamInstallStatus?.length > 0 && streamText === 'enabled' && upgradable !== true):
+        return <><CheckIcon color="green" /> {INSTALLED_STATE.UPTODATE}</>;
+      case (streamInstallStatus?.length > 0 && streamText === 'enabled' && upgradable):
+        return <LongArrowAltUpIcon color="blue"> {INSTALLED_STATE.UPGRADEABLE}</LongArrowAltUpIcon>;
+      default:
+        return <InactiveText text={INSTALLED_STATE.NOTINSTALLED} />;
+    }
+  };
+
+  EnabledIcon.propTypes = {
+    streamText: PropTypes.string.isRequired,
+    streamInstallStatus: PropTypes.arrayOf(PropTypes.string).isRequired,
+    upgradable: PropTypes.bool.isRequired,
+  };
+
+  const StreamState = ({ moduleStreamStatus }) => {
+    let color;
+    let labelVariant;
```

Comment:
I think we can now just put your values directly within the <Label variables below, as you only have 3 different outcomes.